src/main/java/org/web3j/tx/ManagedTransaction.java

```diff
@@ -75,8 +81,15 @@ protected TransactionReceipt processResponse(EthSendTransaction transactionRespo
         }
 
         String transactionHash = transactionResponse.getTransactionHash();
-
-        return waitForTransactionReceipt(transactionHash);
+        //if attempts is zero, means the transaction will not wait for a receipt,
+        if (attempts > 0) {
+            return waitForTransactionReceipt(transactionHash);
+        } else {
+            //but returns an incomplete receipt with only the transactionHash in it
```

Comment:
I think put this in a separate method "emptyTransactionReceipt" it makes the intention clearer.